addons/nuxeo-binary-metadata/src/main/java/org/nuxeo/binary/metadata/internals/BinaryMetadataServiceImpl.java

```diff
@@ -252,16 +253,18 @@ public void handleUpdate(List<MetadataMappingDescriptor> mappingDescriptors, Doc
             if (blob != null) {
                 boolean isDirtyMapping = isDirtyMapping(mappingDescriptor, doc);
                 if (isDirtyMapping) {
-                    BlobManager blobManager = Framework.getService(BlobManager.class);
-                    BlobProvider blobProvider = blobManager.getBlobProvider(blob);
-                    // do not write metadata in blobs backed by extended blob providers (ex: Google Drive) or blobs from
-                    // providers that prevent user updates
-                    if (blobProvider != null && (!blobProvider.supportsUserUpdate() || blobProvider.getBinaryManager() == null)) {
-                        return;
+                    if(!mappingDescriptor.getReadOnly()) {
```

Comment:
Does it make sense to join the `isDirtyMapping` and `readOnly` conditions? As it we won't loose history and code will be less complex.

Reply:
I wanted to separate the `if` handling the mapping from the `else` that is about the blob.
But if it's better practice to merge them, I'm ok with that of course

Reply:
It depends on the case. Globally if we can join the ifs we do it.
Here, under the scenario below during an update:
- update some metadata which are under the binary mapping
- the mapping is read only
- update the blob which is under the binary mapping

Without the join, we do nothing whereas with it, we update the document with metadata coming from blob. So updated metadata could be erased. I think this is not what we want? But, on the other hand, if we just update the document's blob, these metadata will be updated (if we join the two conditions).

In the end, question is more: Do we want to keep binary metadata in sync with their blob when readOnly flag is true?

Reply:
So, no actually. When readonly is true we want to:
-> Update the fields only if the blob changes
-> Do nothing to the blog if the fields change.

So ultimately, blob metadata and fields can end up being desynchronized (user bodies a field). But if the blob is modified, then all fields are reset.

I mean, we don't keep track of changes. We don't do this: "If field1 was modified by a user, we don't modify the field if a new blob is uploaded". This could be another feature, later :-)

Reply:
Sure. 👍

Reply:
Also, while re-reviewing it, I did this also to better stick to the comment of the interface (`BinaryMetadataService#handleUpdate`}, so we clearly separate a change in the metadata from a change in the blob (and I forgot to update the comment of these methods to briefly talk about the readOnly flag.

So, ok to keep it with 2 `if`?

Reply:
It's ok 😉